www/source/blog/2018-08-18-lift-and-shift-hab.html.md

```diff
@@ -0,0 +1,396 @@
+* Source control with `git`
+* Python language installation
+* Virtual environment setup
+* Python packaging with `pip` 
+* Internal application structure e.g. where to find the thing to run
+
+A similar list could likely be formulated for applications written in other languages.  In any case, this complicates life for performing any changes at a later stage e.g.
+* Switching to a MySQL database backend
+* Running on a new machine, or
+* Moving to a (new) Cloud provider or service
+
+
+## Packaging Table Setting with Habitat
+
+The following directory structure within the Table Setting repository relates to Habitat packaging:
+
+```bash
+$ tree habitat/
+habitat/
+├── README.md
+├── config
+├── default.toml
+├── hooks
+│   ├── init
+│   └── run
+└── plan.sh
+```
+This incorporates a set of steps to install required dependencies, similar to those taken when running locally and also includes the MySQL library setup.  Table Setting explicitly depends on a known good Python version. Environment variables are used to convey runtime configuration to Table Setting, some are conditionally set when e.g. binding to a Habitat managed MySQL database.  
+
+Let's concentrate for now on creating deployment artifacts with Habitat.  Below we use the Studio to export different package formats:
+```bash
+$ cd table-setting
+$ hab studio enter
+[STUDIO] build
+[STUDIO] hab pkg export tar $(ls -1t results/*.hart | head -1)
+[STUDIO] hab pkg export docker $(ls -1t results/*.hart | head -1)
+```
+
+We also use the Habitat maintained `core/mysql` image for some deployments, let's install and export that locally also:
+
+```bash
+[STUDIO] hab pkg install core/mysql
+» Installing core/mysql
+☁ Determining latest version of core/mysql in the 'stable' channel
+☛ Verifying core/mysql/5.7.21/20180609181124
+... output curtailed
+★ Install of  core/mysql/5.7.21/20180609181124 complete with 25 new packages installed.
+```
+
+Note the package version from the installation command, we will now use this to export a Docker image locally:
+
+```
+[STUDIO] hab pkg export docker core/mysql/5.7.21/20180609181124
+```
+
+For reference, all of the plans built and maintained by the Habitat Core Team are [available here](https://github.com/habitat-sh/core-plans).
+
+The resulting artifacts are all we need for the deployment scenarios below.
+
+## Bare metal or standalone VM deployment
+
+Let's assume we want to deploy Table Setting to an Ubuntu machine and use a file-based SQLite database backend.  In a real-world setup the versioned `tar` deployment file might be stored in a dedicated artifact repository but we will assume it is locally available on the target machine here.
+
+Relying on the Habitat package naming convention, we can run the package as follows:
+
+```bash
+$ cat hab_tar_deployment_ubuntu.sh
+#!/bin/bash
+# one-time setup - packages can specify a linux user to run under, by default this is the hab user 
```

Comment:
"packages can specify a linux user to run under, by default this is the hab user" -> "packages can specify a Linux user to run under; by default this is the `hab` user "